design/idempotency_ids.md

```diff
@@ -76,13 +85,29 @@ If a transaction learns that it has been in-flight so long that its idempotency
 
 # Considerations
 
-- Additional storage space on the cluster. This can be controlled directly via an idempotency id target bytes knob/config.
+- Additional storage space on the cluster.
 - Potential write hot spot.
 
 # Multi-version client
 
 The multi-version client will generate its own idempotency id for a transaction and manage its lifecycle. It will duplicate the logic in NativeApi to achieve the same guarantees. As part of this change we will also ensure that the previous commit attempt is no longer in-flight before allowing the commit future to become ready. This will fix a potential "causal-write-risky" issue if a commit attempt fails with `cluster_version_changed`.
 
+# Implementation
+
+```
+FDBFuture* fdb_database_commit_result(FDBDatabase* db, uint8_t const* idempotency_id, int idempotency_id_length, int64_t read_snapshot)
+```
+
+The implementation would first commit a transaction that conflicts with any transaction with the same idempotency id (recall that the idempotency id is added to the conflict ranges), to ensure that none are in flight.
```

Comment:
This sentence is a bit unclear. If I understood it correctly, here's a potential rephrase:

The implementation will commit a dummy transaction that conflicts with transactions that have this `idempotency_id`. (Recall that `idempotency_id` is added to the conflict ranges, so any transaction with the same idempotency id will have the same conflict ranges.) This ensures that no other conflicting transaction are in flight.

Reply:
Rephrased